### PR TITLE
Update ug

### PR DESCRIPTION
**Pull request by Harry-Gao-H**
Add Table of Content and Command Navigation.
Change the order of applicant command and position command because the users need to first add positions.

### CODE REVIEW

**Review by BenjaminTan99 — APPROVED** [review]
LGTM. Nice changes to UG.

File: docs/UserGuide.md [hunk 1]
```diff
@@ -58,6 +68,98 @@ MrTechRecruiter (MTR) is a **desktop app for managing job applications, optimize
...
 
+## Command Navigation
+
+
+### General Commands
+* Viewing help: [`help`](#viewing-help--help)
+* Undoing the last modification: [`undo`](#undoing-the-last-modification--undo)
+* Exiting the program: [`exit`](#exiting-the-program--exit)
+
+### Position-Related Command
```
Comment (LMAOboxhack) [inline docs/UserGuide.md:79]:
```suggestion
### Position-Related Commands
```

File: docs/UserGuide.md [hunk 2]
```diff
@@ -58,6 +68,98 @@ MrTechRecruiter (MTR) is a **desktop app for managing job applications, optimize
...
+* Adding a position: [`add-position`](#adding-a-position-add-position)
+* Deleting a position: [`delete-position`](#deleting-a-position-delete-position)
+* Editing a position: [`edit-position`](#editing-a-position-edit-position)
+* Listing all positions : [`list-position`](#listing-all-positions--list-position)
+* Visualizing a position with a pie chart: [`visualize`](#visualizing-a-position-with-a-pie-chart-visualize)
+* Get rejection rate of a specified position : [`rate`](#get-rejection-rate-of-a-specified-position--rate)
+
+### Applicant-Related Command
```
Comment (LMAOboxhack) [inline docs/UserGuide.md:87]:
```suggestion
### Applicant-Related Commands
```

File: docs/UserGuide.md [hunk 3]
```diff
@@ -93,7 +195,7 @@ Format: `edit-applicant INDEX [n/NAME p/PHONE_NUMBER e/EMAIL a/ADDRESS pos/POSIT
 
 * Edit the applicant with the specified `INDEX`
 * At least one argument should be present
-* The `position` must have been added to MrTechRecruiter
+* If `position` field is edited, the new `position` must have been added to MrTechRecruiter
```
Comment (LMAOboxhack) [inline docs/UserGuide.md:198]:
```suggestion
* If the `position` field is edited, the new `position` must already exist in MrTechRecruiter
```

**Review by LMAOboxhack — APPROVED** [review]
Minor nits, LGTM

**Pull request merged**